### PR TITLE
ZD#1290963 Handle case where the next hour is in the next day

### PR DESCRIPTION
**Pull request by hsume2**
/cc @jcheatham @zendesk/twrk @zendesk/sustaining

### CODE REVIEW

**Comment by alex-stone** [comment]
:+1: on :green_apple:

File: lib/biz/time.rb [hunk 1]
```diff
@@ -39,7 +40,12 @@ def on_date(date, day_time)
         true
       )
     rescue TZInfo::PeriodNotFound
-      on_date(date, DayTime.new(day_time.day_second + HOUR))
+      next_day_second = day_time.day_second + HOUR
+      if next_day_second > SECONDS_IN_DAY
+        on_date(date + 1, DayTime.new(next_day_second % SECONDS_IN_DAY))
```
Comment (jcheatham) [inline lib/biz/time.rb]: Do we know it's going to be 86400 + 3600 here?  Can just say `DayTime.new(HOUR)` here?

**Comment by jcheatham** [comment]
:+1:

File: lib/biz/time.rb (hunk 1 above)
Comment (hsume2) [inline lib/biz/time.rb]: @jcheatham all we know is that there are local times that have no UTC equivalent ([ref](http://www.rubydoc.info/gems/tzinfo/TZInfo/Timezone#local_to_utc-instance_method)), we can't be guaranteed that it's necessarily `86400`.

Comment (jcheatham) [inline lib/biz/time.rb]: yeah, wasn't sure whether this could be simplified to drop the modulus, but this is undoubtedly the safest :)

**Pull request merged**